### DISCUSSION TITLE
Idea: break up binary distributions

### DISCUSSION THREAD

**Discussion by olanti-p** [cataclysmbn/Cataclysm-BN#3301] [General]
This idea was brought up on Discord some time ago, and it's essentially aimed at opimizing download sizes. 

Since tilesets rarely change, if at all, but take up a lot of space in git history as binary blobs and in compressed releases, they can be moved entirely into some external repository, to be downloaded by players from there.
This allows decreasing download size when updating the game, but on the other hand complicates installation steps for end-users, especially on platforms that don't take kindly to launchers (Android, OSX). A middle ground could be to have CI build separate distributions, with and without tilesets.

**Comment by chaosvolt** [comment]
The main concern I see with moving tilesets to a separate repository is player confusion, as they'll likely not realize at first where to grab the tilesets and that they need to do so to use them. It'll also add even more complications given we're already got several different versions of UDP floating around (unpacked and packed not always being in sync with each other plus being developed mainly for DDA).

Catapult launcher might make it easier but they'd have to be kept in the loop and told to link to those repos, plus most launcher users will likely fail to realize that the tileset "mods" will become effectively mandatory.

**Comment by scarf005** [comment]
we could still package tilesets but download them from external sources when creating new release.
